pkg/deviceshifu/deviceshifubase/pushtelemetry.go

```diff
@@ -159,8 +159,12 @@ func injectSecret(c *rest.RESTClient, ts *v1alpha1.TelemetryService, ns string)
 	username, exist := secret[UsernameSecretField]
 	if !exist {
 		logger.Errorf("the %v field not found in telemetry secret", UsernameSecretField)
+		if ts.Spec.ServiceSettings.HTTPSetting.Username == nil {
+			ts.Spec.ServiceSettings.HTTPSetting.Username = new(string)
+		}
 	} else {
-		*ts.Spec.ServiceSettings.HTTPSetting.Username = username
+		// get pointer of string
+		ts.Spec.ServiceSettings.HTTPSetting.Username = struct{ str *string }{str: &username}.str
```

Comment:
ts.Spec.ServiceSettings.HTTPSetting.Username = &username

Reply:
Also I think *ts.Spec.ServiceSettings.HTTPSetting.Username = username is also correct, not sure why you changed it

Reply:
> Also I think *ts.Spec.ServiceSettings.HTTPSetting.Username = username is also correct, not sure why you changed it

Because the user may write username to secret and thus delete the field in HTTPSetting, so HTTPSetting.Username may be nil here.